draft-ietf-quic-transport.md

```diff
@@ -7486,6 +7483,33 @@ ECN to be disabled.  For those rare cases where marked packets are discarded by
 the path, the short duration of the testing period limits the number of losses
 incurred.
 
+# Sample Variable-Length Integer Decoding {#sample-varint}
+
+The pseudo-code in {{alg-varint}} shows how a variable-length integer can be
+read from a stream of bytes.  The function ReadVarint takes a single argument, a
+sequence of bytes which can be read in network byte order.
+
+~~~
+ReadVarint(data):
+  // The length of variable-length integers is encoded in the
+  // first two bits of the first byte.
+  v = data.next_byte()
+  length = (1 << (v >> 6))
+
+  // Once the length is known, remove these bits and read any
+  // remaining bytes.
+  v = v & 0x3f
+  repeat length-1 times:
+    v = (v << 8) + data.next_byte()
+  return v
```

Comment:
`v` is also in network byte-order, no?

Reply:
Yes, but I'm not sure that's necessary to mention.  We haven't changed it over the course of the algorithm.

Reply:
How could it not be in network byte order? Unless next_byte does something other than read the next byte.

Reply:
What I meant is that we might want to say that the value still needs to co vetted into host byte order before it can be used.

Reply:
Doesn't this do that? I mean, the operations here are numeric, so unless `<<` means "divide by 2^n", we're good, right?

Reply:
Yeah -- there's nothing here that is endian-specific.